extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1630,8 +1635,14 @@ private void checkPendingCompletionTasks() throws ExecutionException, Interrupte
 
           // reset partitions offsets for this task group so that they will be re-read from metadata storage
           partitionGroups.get(groupId).replaceAll((partition, offset) -> NOT_SET);
-          sequenceTaskGroup.remove(generateSequenceName(groupId));
-
+          if (taskGroups.get(groupId) == null) {
```

Comment:
I think if `generateSequenceName` took a TaskGroup object, you could just pass in `group` and not worry about a null check. It won't be null, since it was pulled out of `taskGroupList` earlier.

Reply:
that makes sense.